tests/integration/test_code.py

```diff
@@ -0,0 +1,471 @@
+
+            # If the line ends with an elapsed time, replace it with something constant.
+            line = re.sub(r"\d+:\d+:\d+$", "H:MM:SS", line)
+            # In the progress bar for syncing vscode extensions, there might be one with
+            # N/N (which depends on how many extensions were missing). Replace it with a
+            # constant.
+            line = re.sub(r" \d+/\d+ ", " N/N ", line)
+
+            return (
+                line.rstrip()
+                .replace(str(job_id), "JOB_ID")
+                .replace(node_hostname, "COMPUTE_NODE")
+                .replace(home, "$HOME")
+                .replace(
+                    f"--account={slurm_account_on_cluster}", "--account=SLURM_ACCOUNT"
+                )
+            )
+
+        return "\n".join(filter_line(line) for line in captured_output.splitlines())
+
+    file_regression.check(filter_captured_output(captured_output))
+
+
+@pytest.mark.parametrize("use_v1", [False, True], ids=["code", "code_v1"])
+@pytest.mark.asyncio
+async def test_code_without_code_command_in_path(
+    monkeypatch: pytest.MonkeyPatch, use_v1: bool
+):
+    """Test the case where `mila code` is run without having vscode installed."""
+
+    def mock_which(command: str) -> str | None:
+        assert command == "code"  # pretend like vscode isn't installed.
+        return None
+
+    monkeypatch.setattr(shutil, shutil.which.__name__, Mock(side_effect=mock_which))
+    if use_v1:
+        with pytest.raises(CommandNotFoundError):
+            code_v1(
+                path="bob",
+                command="code",
+                persist=False,
+                job=None,
+                node=None,
+                alloc=[],
+                cluster="bob",
+            )
+    else:
+        with pytest.raises(CommandNotFoundError):
+            await code(
+                path="bob",
+                command="code",
+                persist=False,
+                job=None,
+                node=None,
+                alloc=[],
+                cluster="bob",
+            )
+
+
+@pytest.fixture(scope="session")
+def persistent_job_name(job_name: str) -> str | None:
+    """Name of a job that should be used for tests and not be cancelled when exiting."""
+    return job_name + "_persistent"
+
+
+@pytest_asyncio.fixture(scope="session")
+async def existing_job(
+    cluster: str,
+    login_node_v2: RemoteV2,
+    allocation_flags: list[str],
+    job_name: str,
+    persistent_job_name: str,
+) -> ComputeNode:
+    """Gets a compute node connecting to a running job on the cluster.
+
+    This avoids making an allocation if possible, by reusing an already-running job
+    with the name `persistent_job_name` or `job_name`, or any running job if found.
+
+    This will try to use either persistent or temporary jobs (small jobs that are used
+    across test runs, or jobs that are spawned just for a single test run) to avoid
+    making a new allocation on the cluster.
+    """
```

Comment:
Is there a change a non-persistent job could die before the end of the tests? Otherwise, what would be the difference between the persistent and non-persistent job?

Reply:
I simplified this slightly in 27fbd5ad86045a8e9aa24c86aada617373735f9e, removing the idea of "persistent" jobs.
This will now try to reuse a running test job if one is found. If not, then it will spawn a new job with `salloc`.
If we wanted to, we could also just make a new job allocation per test run and not try to reuse test jobs at all, but this seems a bit wasteful to me.